portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,14 +46,26 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     int maxCommentDisplay = Integer.parseInt(getFieldFromResponse(request, "maxcomments", defaultMaxComment));
+    String sortMetric = getFieldFromResponse(request, "metric", "time");
+    String sortOrder = getFieldFromResponse(request, "order", "des");
 
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
-    Query<Entity> query =
+    Query<Entity> query;
+    if (sortOrder.equals("des")) {
```

Comment:
Instead of creating two Builders here, you can have one Builder and set different attributes. 
i.e. builder = Query.newEntityQueryBuilder()
...set common attributes...
if (condition) {
   builder.setOrderBy(...)
} else {
   builder.setOrderBy(...)
}
query = builder.build();